components/gcp-click-to-deploy/src/DeployForm.tsx

```diff
@@ -482,6 +483,35 @@ export default class DeployForm extends React.Component<any, DeployFormState> {
     }, 10000);
   }
 
+  private _redirectToKFDashboard(dashboardUri: string) {
+    // relying on JupyterHub logo image to be available when the site is ready.
+    // The dashboard URI is hosted at a domain different from the deployer
+    // app. Fetching a GET on the dashboard is blocked by the browser due
+    // to CORS. Therefore we use an img element as a hack which fetches
+    // an image served by the target site, the img load is a simple html
+    // request and not an AJAX request, thus bypassing the CORS in this
+    // case.
+    const imgUri = dashboardUri + 'hub/logo';
+    const startTime = new Date().getTime() / 1000;
+    const img = document.createElement('img');
+    img.src = imgUri + '?rand=' + Math.random();
+    img.id = 'ready_test';
+    img.onload = () => { window.location.href = dashboardUri; };
+    img.onerror = () => {
+      const timeSince = (new Date().getTime() / 1000) - startTime;
```

Comment:
Can we limit request frequency to like once per 10 seconds?
Just to avoid any rate limit.

Reply:
Since its html load of the image, we don't have control over the loading. The browser itself controls the fetch of the image. Generally, the default timeout on a resource fetch by the browser is usually in minutes. For eg. in Chrome, it will wait 5min before it times out and raises an error.

Reply:
The DNS record is not up on initial minutes, so img load would fails immediately and keep busy waiting till DNS up. We might have huge amount of log output and img load requests.